sdks/python/apache_beam/runners/dataflow/dataflow_metrics.py

```diff
@@ -196,17 +219,68 @@ def _get_metrics_from_dataflow(self):
 
     job_metrics = self._dataflow_client.get_job_metrics(job_id)
     # If the job has terminated, metrics will not change and we can cache them.
-    if self.job_result.is_in_terminal_state():
+    if self.job_result and self.job_result.is_in_terminal_state():
       self._cached_metrics = job_metrics
     return job_metrics
 
+  def all_metrics(self):
+    """Return all user and system metrics from the dataflow service."""
+    metric_results = []
+    response = self._get_metrics_from_dataflow()
+    self._populate_metrics(response, metric_results, user_metrics=True)
+    self._populate_metrics(response, metric_results, user_metrics=False)
+    return metric_results
+
   def query(self, filter=None):
+    metric_results = []
     response = self._get_metrics_from_dataflow()
-    metric_results = self._populate_metric_results(response)
+    self._populate_metrics(response, metric_results, user_metrics=True)
     return {self.COUNTERS: [elm for elm in metric_results
                             if self.matches(filter, elm.key)
                             and DataflowMetrics._is_counter(elm)],
             self.DISTRIBUTIONS: [elm for elm in metric_results
                                  if self.matches(filter, elm.key)
                                  and DataflowMetrics._is_distribution(elm)],
             self.GAUGES: []}  # TODO(pabloem): Add Gauge support for dataflow.
+
+
+def main(argv):
+  """Print the metric results for a the dataflow --job_id and --project.
+
+  Instead of running an entire pipeline which takes several minutes, use this
+  main method to display MetricResults for a specific --job_id and --project
+  which takes only a few seconds.
+  """
+  # Import here to avoid adding the dependency for local running scenarios.
+  try:
+    # pylint: disable=wrong-import-order, wrong-import-position
+    from apache_beam.runners.dataflow.internal import apiclient
+  except ImportError:
+    raise ImportError(
+        'Google Cloud Dataflow runner not available, '
+        'please install apache_beam[gcp]')
+  if argv[0] == __file__:
+    argv = argv[1:]
```

Comment:
Did you find you had to do this? AFAIK argparse is able to deal with this. - https://docs.python.org/3/library/argparse.html#prog

Reply:
found this pattern in another file.  if argpasrse handles it ill remove it

Reply:
seems to work without it